packages/assets-controllers/src/TokensController.ts

```diff
@@ -905,12 +905,13 @@ export class TokensController extends BaseController<
     this.update({ ignoredTokens: [], allIgnoredTokens: {} });
   }
 
-  _requestApproval(
+  async _requestApproval(
     suggestedAssetMeta: SuggestedAssetMeta & {
       interactingAddress: string;
     },
+    shouldShowRequest = true,
```

Comment:
Is it worth using an optional object to avoid the ambiguous booleans, even if the name remains?

Reply:
Done.